test/functional/ui/live_substitution_spec.lua

```diff
@@ -0,0 +1,511 @@
+          ~                                       |
+          ~                                       |
+          ~                                       |
+          ~                                       |
+          ~                                       |
+          ~                                       |
+          ~                                       |
+          ~                                       |
+          :%s/are/ARE                             |
+    ]])
+    end)
+
+    it('no split if :s', function()
+        insert([[
+      these are some lines
+      with colorful text (are)]])
+        feed(':s/are/ARE')
+
+        screen:expect([[
+      these are some lines                    |
+      with colorful text (ARE)                |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      :s/are/ARE^                              |
+    ]])
+    end)
+
+    it('split if :%s/are', function()
+        insert([[
+      these are some lines
+      without colorful text (are)]])
+        execute('set hlsearch')
+        feed(':%s/are')
+
+        screen:expect([[
+        these {UNEXPECTED background = Screen.colors.Yellow:are} some lines                    |
```

Comment:
`UNEXPECTED` shouldn't be in a test, and should not ever be emitted by `snapshot_util()` ( If it is please file a bug), I guess it was manually edited but it is better to just add `snapshot_util()` again before the test and replace it (and optionally append the new groups to `set_default_attr_ids()`)